core/src/main/java/ch/cyberduck/core/ProtocolFactory.java

```diff
@@ -220,33 +222,101 @@ public Protocol forName(final String identifier, final String provider) {
      * @return Matching protocol or null if no match
      */
     public Protocol forName(final List<Protocol> enabled, final String identifier, final String provider) {
-        final Protocol match =
+        int state = 0; // Acts as findFirst-guard
+        Protocol match = null;
+        final Iterable<Protocol> protocols = enabled.stream().sorted(new DeprecatedProtocolComparator())::iterator;
+        for(final Protocol protocol : protocols) {
```

Comment:
As in https://github.com/iterate-ch/cyberduck/pull/15819/files#r1559353773, but:
As Streams cannot be reused, make one loop over all enabled protocols, and perform comparisons once.
Will only perform comparisons if applicable (state gated).